File: dcp-solutions/2024-11-06/solution.rb
```ruby
def break_string_into_lines(s, k):
  words = s.split
  lines = []
  current_line = []
  line_length = 0

  until words.empty?
    word = words.shift
    if line_length + word.length + 1 <= k
      current_line << word
      line_length += word.length + 1
    else
      lines << current_line.join(' ')
      current_line = [word]
      line_length = word.length + 1
    end
  end

  lines << current_line.join(' ') if !current_line.empty?

  return lines unless line_length > k

  null
end
```

File: dcp-solutions/2024-11-06/solution.php
```php
<?php
function breakString($s, $k) {
    $words = explode(' ', trim($s));
    if (empty($words)) return null;

    $result = [];
    $currentLine = '';
    foreach ($words as $word) {
        if (strlen($currentLine) + strlen($word) + 1 > $k) {
            if (!empty($currentLine)) array_push($result, $currentLine);
            $currentLine = $word;
        } else {
            if ($currentLine) $currentLine .= ' ';
            $currentLine .= $word;
        }
    }
    if ($currentLine) array_push($result, $currentLine);

    // If the number of lines in result is more than 1 and each line has only one word then return null
    $lines = count($result);
    for ($i=0; $i<$lines ; $i++) {
        if (count($result[$i]) == 1) return null;
    }
    return $result;
}
```